lucien-ios/LoginScreen/LoginViewModel.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  LoginManager.swift
+//  lucien-ios
+//
+//  Created by Fang, Gracie on 9/29/17.
+//  Copyright © 2017 Intrepid Pursuits. All rights reserved.
+//
+
+final class LoginViewModel {
+
+    let lucienAPIClient = LucienAPIClient()
+    var currentUser: User?
+    var dashboardData: Dashboard?
+    var hasCollection: Bool = false
+
+    func getCurrentUser(completion: @escaping () -> Void) {
+        lucienAPIClient.getCurrentUser { response in
+            switch response {
+            case .success(let user):
+                self.currentUser = user
+                completion()
+            case .failure(let error):
+                print(error)
+            }
+        }
+    }
+
+    func authenticateUser(code: String, completion: @escaping () -> Void) {
+        lucienAPIClient.authenticateUser(code: code) { response in
+            switch response {
+            case .success(let authenticationResponse):
+                LucienRequest.authorizationHeader = "Bearer " + authenticationResponse.token
+                completion()
+            case .failure(let error):
+                print(error)
+            }
+        }
+    }
+
+    func getDashboard(completion: @escaping () -> Void) {
```

Comment:
Why does this have a completion? Also if you're going to have a completion, you need to call it from all exit paths (it's not being called in the failure case). This applies to all completions, in this file and in general